ferry.go

```diff
@@ -215,6 +217,8 @@ func (f *Ferry) NewIterativeVerifier() (*IterativeVerifier, error) {
 		TargetDB:            f.TargetDB,
 		CompressionVerifier: compressionVerifier,
 
+		Tables:              f.Tables.AsSlice(),
+		TableSchemaCache:    f.Tables,
```

Comment:
It makes me uneasy to be suggesting this but do you think it's a good idea to return an error if one of these `Ferry.NewThing()` constructors are called before the Ferry is initialized? (perhaps checked with a `Ferry.initialized` boolean field)? 

The ultimate cleanup would be that we get rid of `Ferry.Initialize` in favour of a `NewFerry()` constructor but that's long ways away from our current state

Reply:
the `Ferry.NewComponent()` methods, except for the IterativeVerifier, only returns the component structs being created. Adding an error would make the methods more annoying to use.

What if I just `panic` if the Ferry is uninitialized and these methods are called? If such a code was written, it's guarenteed to be a programming error as the resulting components won't even have a valid `SourceDB` and such. This would basically match the current behaviour: components created before Ferry is initialized will likely panic due to nil pointers.


(I agree with the spirit of `NewFerry`, but I don't know how to create a sane version of it without a large amount of positional arguments.)

Reply:
the `panic` is a good enough measure for now. Add a `TODO: refactor Ferry.Initialize to NewFerry constructor` right by it and we're good to go